src/site/content/en/blog/layout-instability-api/index.md

```diff
@@ -0,0 +1,331 @@
+  explains how to use the API and provide feedback
+tags:
+  - post # post is a required tag for the article to show up in the blog.
+  - performance
+  - origin-trials
+---
+
+Have you ever been reading an article online when something suddenly changes on
+the page? Without warning, the text moves, and you've lost your place.
+
+Or even worse: you're about to tap a link or a button, but in the instant
+before your finger lands&mdash;BOOM!&mdash;the link moves, and you end up
+clicking something else!
+
+Most of the time these kinds of experiences are just a mild annoyance or
+frustration. But in some cases, they can actually cause real damage or harm.
+
+Unexpected movement of page content usually happens because resources are
+loaded asynchronously or DOM elements get dynamically added to the page above
+existing content. The culprit might be an image or video with unknown
+dimensions, a font that renders larger or smaller than its fallback, or a
+third-party ad or widget that dynamically resizes itself.
+
+What makes this issue even more problematic is that how a site functions in
+development is often quite different from how users experience it in
+production: personalized or third-party content often doesn't behave the same
+in development as it does in production, test images are often already in the
+developer’s browser cache, and API calls that run locally are often so fast
+that the delay isn't noticeable.
+
+The first step toward properly solving this problem is to give developers the
+tools to measure it and understand how often it's occurring for real users. The
+[Layout Instability API](https://github.com/WICG/layout-instability), currently
+being incubated in the [WICG](https://www.w3.org/community/wicg/), aims to
+address this.
+
+## How is layout instability determined?
+
+Layout Instability is determined by calculating a _layout shift_ score every
+time the browser renders a new frame. Developers and analytics providers can
+then monitor these scores in the wild, identify the culprits, and improve the
+experience for their users.
+
+### Layout shift defined
+
+For each element that's visible in the viewport, if the element's block-start
+and inline-start positions (e.g., its top and left position in the default
+[writing mode](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode))
+relative to its [containing
+element](https://www.w3.org/TR/CSS2/visudet.html#containing-block-details) has
+changed between the frame currently being rendered and the frame that was
+previously rendered, the element is identified as unstable.
+
+The union of the visible areas of all unstable elements for the previous frame
+_and_ the current frame&mdash;as a fraction of the total area of the
+viewport&mdash;is the **layout shift score** for the current frame.
+
+Here are a few examples of how the layout shift score is calculated:
+
+![Layout shift example #1](layout-shift-1.png)
```

Comment:
For all of the image examples, would you mind enhancing the alt text a bit? If you were describing the image to someone over the phone how would you phrase it?